### ISSUE TITLE
JunitMustMatchers strange behaviour when using contains

### ISSUE THREAD

**Issue by greenjoe** [etorreborre/specs2#653]
Hello!
I have a specs2 test 
```

import org.specs2.matcher.{Matcher, MustMatchers}
import org.specs2.mutable.SpecWithJUnit

class SpecsTest extends SpecWithJUnit with MustMatchers{
  "SpecsTest" should {
    "work for a list" in {
      val l = List(1, 2, 3)

      val match3: Matcher[Any] = beLike[Any] {
        case x: Int => x === 3
      }

      l must contain(match3)

    }
  }
}
```
it passes and works fine. I would like to have the same test written with `JUnitMustMatchers`, so I wrote
```
import org.junit.Test
import org.specs2.matcher.{JUnitMustMatchers, Matcher}

class JUnitTest extends JUnitMustMatchers {

  @Test
  def myTest() {
    val l = List(1, 2, 3)

    val match3: Matcher[Any] = beLike[Any] {
      case x: Int => x === 3
    }

    l must contain(match3)

  }
}


```
Unfortunately it does not pass. 
```
org.specs2.execute.ErrorException: org.specs2.matcher.JUnitExpectations$$anon$1: 1 != 3 
Expected :3
Actual   :1
	at org.specs2.matcher.ContainWithResult.$anonfun$apply$13(TraversableMatchers.scala:219)
	at scala.collection.TraversableOnce.$anonfun$foldLeft$1(TraversableOnce.scala:157)
```
Am I using `JUnitMustMatchers` incorrectly here for some reason or is it a bug?

**Comment by etorreborre** [comment]
You can find all the jars on sonatype: https://oss.sonatype.org/content/repositories/releases/org/specs2/specs2-matcher_2.12/4.2.0-2a08d4cae-20180427164428. This repository is generally synchronized with the main maven repository but there can be a delay.

**Comment by greenjoe** [comment]
Okay :-) It looks like it managed to make its way to maven central. I tested with 

```
       <dependency>
            <groupId>org.specs2</groupId>
            <artifactId>specs2-core_2.12</artifactId>
            <version>4.2.0-2a08d4cae-20180427164428</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.specs2</groupId>
            <artifactId>specs2-junit_2.12</artifactId>
            <version>4.2.0-2a08d4cae-20180427164428</version>
            <scope>test</scope>
        </dependency>
```

But it is still not working (in the same way). I'm attaching my Intellij Idea / Maven test project on which I'm trying it out. 
[testspecsjunit.zip](https://github.com/etorreborre/specs2/files/1957393/testspecsjunit.zip)

**Comment by etorreborre** [comment]
Ok that required an additional small change, please try `4.2.0-add90d25e-20180428143534`. And thank you for the small test project, this was immensely helpful!

**Comment by etorreborre** [comment]
By the way you can add this section to your pom file to avoid waiting for the maven central sync:
```
<repositories>
    <repository>
        <id>oss-sonatype</id>
        <name>oss-sonatype</name>
        <url>https://oss.sonatype.org/content/repositories/releases/</url>
        <snapshots>
            <enabled>true</enabled>
        </snapshots>
    </repository>
</repositories>
```

**Comment by greenjoe** [comment]
I confirm that the issue is fixed in version `4.2.0-add90d25e-20180428143534`. Both `must contain` and `must not contain` work fine with `JUnitMustMatchers` in that version. 

Thank you for resolving it so quickly :-) :+1:

**Comment by etorreborre** [comment]
Cool thanks for confirming.
